### PR TITLE
Replace stype with ltype check for validation target column

### PR DESCRIPTION
**Pull request by oleksiyskononenko**
- allow training and validation target columns to have different `stype`s, their `ltype`s must be identical though. Addresses https://github.com/h2oai/h2oai/issues/10593#issuecomment-522372576
- minor improvement of `LType`'s handling.

### CODE REVIEW

File: c/models/py_ftrl.cc [hunk 1]
```diff
@@ -331,9 +331,14 @@ oobj Ftrl::fit(const PKArgs& args) {
...
-    if (dt_y_val->get_ocolumn(0).stype() != dt_y->get_ocolumn(0).stype()) {
-      throw ValueError() << "Validation target frame must have the same "
-                            "stype as the target frame";
+
+    SType stype = dt_y->get_ocolumn(0).stype();
+    SType stype_val = dt_y_val->get_ocolumn(0).stype();
+
+    if (info(stype).ltype() != info(stype_val).ltype()) {
+      throw TypeError() << "Training and validation target columns must have "
+                        << "the same ltype, got: `" << info(stype).ltype_name()
```
Comment (st-pasha) [inline c/models/py_ftrl.cc]: it's probably better to add an `Exception::operator<<(LType)`

File: c/models/py_ftrl.cc [hunk 2]
```diff
@@ -331,9 +331,14 @@ oobj Ftrl::fit(const PKArgs& args) {
                          << "one column";
     }
 
-    if (dt_y_val->get_ocolumn(0).stype() != dt_y->get_ocolumn(0).stype()) {
-      throw ValueError() << "Validation target frame must have the same "
-                            "stype as the target frame";
+
+    SType stype = dt_y->get_ocolumn(0).stype();
+    SType stype_val = dt_y_val->get_ocolumn(0).stype();
```
[the remark this answers is not in the document]
Comment (oleksiyskononenko) [reply c/models/py_ftrl.cc]: Yes, I know, but need `stype` for doing `info(stype).ltype_name()` later. Or there is any better way to get `ltype_name()`?

File: c/models/py_ftrl.cc (hunk 1 above)
Comment (oleksiyskononenko) [reply c/models/py_ftrl.cc]: I guess you mean `Error::operator<<(LType)`? Added now.

File: c/models/py_ftrl.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (oleksiyskononenko) [reply c/models/py_ftrl.cc]: Now I don't need `info(stype)`, so this part has been simplified.

**Pull request merged**